src/main/java/org/folio/innreach/util/OffsetBasedPageRequest.java

```diff
@@ -0,0 +1,70 @@
+package org.folio.innreach.util;
+
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+
+public class OffsetBasedPageRequest implements Pageable {
```

Comment:
there is an implementation already to map offset/limit to page/ size, in folio-spring-base:
https://github.com/folio-org/folio-spring-base/blob/0154cacf66cc8550654c16af13e96b03eb9a6e0b/src/main/java/org/folio/spring/data/OffsetRequest.java